ducktape/tests/loader.py

```diff
@@ -70,35 +70,68 @@ def __init__(self, session_context, logger, repeat=1, injected_args=None, cluste
+            directory = os.path.abspath(directory)
+
+            test_context_list_for_symbol = self.discover(directory, module_name, cls_name, method_name,
+                                                         allow_empty=allow_empty)
+            all_test_context_list.update(test_context_list_for_symbol)
+
+            if not allow_empty and len(test_context_list_for_symbol) == 0:
+                raise LoaderException("Didn't find any tests for symbol %s." % symbol)
+
+        return all_test_context_list
+
+    def load(self, included_test_symbols, excluded_test_symbols=None):
         """Recurse through packages in file hierarchy starting at base_dir, and return a list of test_context objects
-        for all discovered tests.
+        for all discovered tests. Skip any test_context object if it's found in excluded_test_symbols.
 
         - Discover modules that 'look like' a test. By default, this means the filename is "test_*" or "*_test.py"
         - Discover test classes within each test module. A test class is a subclass of Test which is a leaf
           (i.e. it has no subclasses).
         - Discover test methods within each test class. A test method is a method containing 'test' in its name
+        - If excluded_test_symbols is provided, those test methods will be excluded from the final list of test_context
+          objects
 
-        :param test_discovery_symbols: list of file paths
+        :param included_test_symbols: list of test symbols
+            (file path, possibly with a ::Class or ::Class.method specified) to include
+        :param excluded_test_symbols: list of test symbols
+            (file path, possibly with a ::Class or ::Class.method specified) to exclude
         :return list of test context objects found during discovery. Note: if self.repeat is set to n, each test_context
             will appear in the list n times.
         """
-        assert type(test_discovery_symbols) == list, "Expected test_discovery_symbols to be a list."
-        all_test_context_list = []
-        for symbol in test_discovery_symbols:
-            directory, module_name, cls_name, method_name = self._parse_discovery_symbol(symbol)
-            directory = os.path.abspath(directory)
+        excluded_contexts_list = self.load_test_contexts(excluded_test_symbols, allow_empty=True)
+        excluded_test_ids = set(map(lambda ctx: ctx.test_id, excluded_contexts_list))
+        included_contexts_list = self.load_test_contexts(included_test_symbols)
 
-            test_context_list_for_symbol = self.discover(directory, module_name, cls_name, method_name)
-            all_test_context_list.extend(test_context_list_for_symbol)
-
-            if len(test_context_list_for_symbol) == 0:
-                raise LoaderException("Didn't find any tests for symbol %s." % symbol)
-
-        self.logger.debug("Discovered these tests: " + str(all_test_context_list))
+        if not included_contexts_list:
```

Comment:
This seems redundant given the checks already happening in `load_test_contexts`.

Reply:
Thanks! Addressed this in the follow up PR.